### ISSUE TITLE
[bug] SearchParameters is not a constructor

### ISSUE THREAD

**Issue by Kikobeats** [algolia/algoliasearch-helper-js#486]
Hello,

From the last release by @bobylito something is broken the `react-instantsearch` build:

```bash
Uncaught TypeError: _algoliasearchHelper.SearchParameters is not a constructor
    at createInstantSearchManager (createInstantSearchManager.js:51)
    at new InstantSearch (InstantSearch.js:80)
    at ReactCompositeComponentWrapper._constructComponentWithoutOwner (ReactCompositeComponent.js:298)
    at ReactCompositeComponentWrapper._constructComponent (ReactCompositeComponent.js:285)
    at ReactCompositeComponentWrapper.mountComponent (ReactCompositeComponent.js:188)
    at Object.mountComponent (ReactReconciler.js:46)
    at ReactCompositeComponentWrapper.performInitialMount (ReactCompositeComponent.js:371)
    at ReactCompositeComponentWrapper.mountComponent (ReactCompositeComponent.js:258)
    at Object.mountComponent (ReactReconciler.js:46)
    at ReactCompositeComponentWrapper.performInitialMount (ReactCompositeComponent.js:371)
```

Tree dependencies:

```bash
├─┬ react-instantsearch@3.2.1
│ ├─┬ algoliasearch@3.21.1
│ │ ├── agentkeepalive@2.2.0
│ │ ├── debug@2.3.3
│ │ ├─┬ envify@4.0.0
│ │ │ ├── esprima@3.1.3
│ │ │ └── through@2.3.8
│ │ ├── es6-promise@4.1.0
│ │ ├── events@1.1.1
│ │ ├── foreach@2.0.5
│ │ ├── isarray@2.0.1
│ │ ├── load-script@1.0.0
│ │ ├── object-keys@1.0.11
│ │ ├── querystring-es3@0.2.1
│ │ ├── reduce@1.0.1
│ │ ├─┬ tap-bail@0.0.0
│ │ │ └── tap-parser@0.4.3
│ │ ├─┬ tap-dot@1.0.5
│ │ │ └─┬ tap-out@1.4.2
│ │ │   ├── re-emitter@1.1.3
│ │ │   ├─┬ readable-stream@2.2.3
│ │ │   │ └── isarray@1.0.0
│ │ │   ├── split@1.0.0
│ │ │   └── trim@0.0.1
│ │ └── tunnel-agent@0.4.3
│ └─┬ algoliasearch-helper@2.20.0
```

It's strange because the dependency looks locked in the [package.json](https://github.com/algolia/instantsearch.js/blob/v2/packages/react-instantsearch/package.json#L20), but not in my local file:

```
  "dependencies": {
    "algoliasearch": "^3.21.1",
    "algoliasearch-helper": "^2.18.1",
    "classnames": "^2.2.5",
    "lodash": "^4.17.4"
  },
```

**Comment by bobylito** [comment]
Thanks @Kikobeats for reporting this issue. I'm looking into it.

**Comment by bobylito** [comment]
@Kikobeats could it be possible that you are now using the esm build?

**Comment by Kikobeats** [comment]
I don't know what is `esm`, but you can see all my code here: https://github.com/windtoday/windtoday-marketplace

**Comment by bobylito** [comment]
Thanks!

**Comment by bobylito** [comment]
So I created an app with create-react-app and it works fine. However I cloned your project and ran it (to be sure) and it is indeed broken.

What I could see is that the build is using the esm build.

<img width="355" alt="screen shot 2017-03-11 at 13 41 04" src="https://cloud.githubusercontent.com/assets/393765/23823342/9fe01796-0660-11e7-8bcc-e5d3dbad8ec4.png">

I will try to figure out why it uses this new build file and if I can give you a quick fix. Sorry for the bad build :(

cc/ @Haroenv @rayrutjes

**Comment by Haroenv** [comment]
I think it should use that build, but it shouldn't break, I'll check later in the afternoon or tonight

**Comment by bobylito** [comment]
@Kikobeats I have a quick fix for you: remove the node_modules folder, force the dependency of `algoliasearch-helper` to 2.19.0 in your project and yarn everything again. Let me know how it works for you.

**Comment by Kikobeats** [comment]
yes this works, I'm going to fix as soon as posible for my production build as workaround

**Comment by rayrutjes** [comment]
It is definitely related to the esm build which is used by default with webpack if available. Looking into it right now. Maybe we can start by simply removing the module and jsnext:main fields from the package.

**Comment by rayrutjes** [comment]
So the issue is that in the es build there is only one default export.
Trying to make sure that we export everything properly right now.

**Comment by rayrutjes** [comment]
I think actually that it is not possible.
It is not safe to convert es5 export to es6.

I think for now the safest is to rollback on this.

**Comment by bobylito** [comment]
Can you make a PR @rayrutjes? I will then release the new version.

**Comment by rayrutjes** [comment]
Yes on it, thanks @bobylito

**Comment by rayrutjes** [comment]
PR ready @bobylito

**Comment by rayrutjes** [comment]
So I think that the conclusion is the following:
Giving that in es5 exports are not statically analyzable, there is no way to provide a es2015 module without creating a custom wrapper that would use static es6 exports.

For now, the safe next move is to remove the es2015 module.

Sorry about the noise, this is my mistake here.

**Comment by bobylito** [comment]
It doesn't matter who is responsible (and I did accept the PR and released it in the first place). Thanks @rayrutjes for being reactive on the issue!

**Comment by bobylito** [comment]
@Kikobeats the new version has been released as 2.20.1. Thanks for your patience.

**Comment by Kikobeats** [comment]
thanks guys 👍
